src/open_r1/rewards.py

```diff
@@ -148,3 +148,51 @@ def cosine_scaled_reward(completions, solution, **kwargs):
         return rewards
 
     return cosine_scaled_reward
+
+
+def get_repetition_penalty_reward(ngram_size: int, max_penalty: float):
+    if max_penalty > 0:
+        raise ValueError(f"max_penalty {max_penalty} should not be positive")
+
+    if max_penalty == 0:
+        return 0
+
+    def zipngram(text: str, ngram_size: int):
+        words = text.lower().split()
+        return zip(*[words[i:] for i in range(ngram_size)])
+
+    def repetition_penalty_reward(completions, *args, **kwargs):
```

Comment:
Since this function is so simple, shall we ditch `args` and `kwargs` in favour of being more explicit? That also means we don't need a docstring :)

Reply:
```suggestion
    def repetition_penalty_reward(completions, **kwargs):
```

Reply:
I have kept `**kwargs` as other reward functions use other arguments and the GRPOTrainer call is not aware of the signature of the function it is calling so provides the additional kwargs, regardless of whether they are used or not.